test/TokenCuratedRegistry.test.js

```diff
@@ -1,26 +1,44 @@
+const { shouldFail, constants } = require('lk-test-helpers')(web3)
 const { shouldBehaveLikeTokenCuratedRegistry } = require('./TokenCuratedRegistry.behavior')
 const { shouldBehaveLikeTimelockableItemRegistry } = require('./TimelockableItemRegistry.behavior')
 const { shouldBehaveLikeStakedRegistry } = require('./StakedRegistry.behavior')
 const { shouldBehaveLikeOwnedItemRegistry } = require('./OwnedItemRegistry.behavior')
 const { shouldBehaveLikeBasicRegistry } = require('./BasicRegistry.behavior')
 
 const TestToken = artifacts.require('TestToken')
+const ChallengeFactory = artifacts.require('MockChallengeFactory')
 const TokenCuratedRegistry = artifacts.require('MockTokenCuratedRegistry')
 
+const { ZERO_ADDRESS } = constants
+
 contract('TokenCuratedRegistry', function (accounts) {
 
-  const [owner, rando] = accounts
+  const [owner, challenger, rando] = accounts
   const initialBalance = 100 * 10 ** 18
   const minStake = 10 * 10 ** 18
+  const mockChallengeReward = minStake * 2 * 75 / 100
   const applicationPeriod = 60 * 60
-  const challengeFactoryAddress = '0x9497F19985AE5e02F7A0dEc7FeE521eaE678d0F7'
 
   beforeEach(async function () {
     this.now = (await web3.eth.getBlock('latest')).timestamp
     this.token = await TestToken.new(
-      [owner, rando],
+      [owner, challenger, rando],
       initialBalance
     )
+    this.challengeFactory = await ChallengeFactory.new(mockChallengeReward)
+  })
+
+  describe('when challenge factory address is zero', function () {
```

Comment:
what separates the tests in here from the tests in the `behavior` files?

Reply:
the way the behavior files are set up, the contracts are instantiated outside of the behavior file and the behavior file tests all the functionality of the instantiated contract. since this is a test for the constructor, thought it made sense to put it in the file where the contract is instantiated. Reason being, if TokenCuratedRegistry was ever extended and the extending contract used the TokenCuratedRegistry.behavior.js tests, the extending contract might have a constructor that takes different parameters. Good example of this is TokenCuratedRegistry and StakedRegistry